### ISSUE TITLE
After changing the schema, deserializeException occurs in iOS.

### ISSUE THREAD

**Issue by Arosf** [aws-amplify/amplify-flutter#415]
**Describe the bug**
I added a name to User as follows, generated code, did an amplify push, and built the iOS app, but an exception occurred. 
The android app did not throw any exceptions. 

```graphql
type User @model {
  id: ID!
  name: String <-- new
}
```

An exception was thrown when the following code was executed. 

```dart
final user = await datastore.query(
  Amp.User.classType,
  where: Amp.User.ID.eq(userId),
);
```

The error message displayed is as follows. 

```
DataStoreException (DataStoreException(message: The operation couldn’t be completed. (SQLite.Result error 0.), recoverySuggestion: The operation couldn’t be completed. (SQLite.Result error 0.), underlyingException: The operation couldn’t be completed. (SQLite.Result error 0.)))
```

Stack trace when an exception is thrown.

```
#0      AmplifyDataStoreMethodChannel.query
package:amplify_datastore/method_channel_datastore.dart:73
<asynchronous suspension>
#1      AmplifyDataStore.query
package:amplify_datastore/amplify_datastore.dart:71
#2      DataStoreCategory.query
package:amplify_flutter/categories/amplify_datastore_category.dart:52
#3      UserRepository.fetch
package:flutter_template/…/user/user_repository.dart:14
<asynchronous suspension>
#4      _UserPageState._checkAndRegistUser
package:flutter_template/main.dart:262
#5      _UserPageState.initState

```

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[] Android
[x] iOS

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone12 Pro Max (ios simulator)]
 - OS: [e.g. iOS14.4]
 - Version [flutter 1.22.6]

**Comment by haverchuck** [comment]
@Arosf Which version of amplify_datastore are you using?

**Comment by fjnoyp** [comment]
Thanks @Arosf we are able to reproduce this issue on our side and are looking into it.

**Comment by Arosf** [comment]
@haverchuck amplify_datastore version is 0.1.0

**Comment by fjnoyp** [comment]
Hi @Arosf the issue you encountered comes from the underlying Amplify iOS Library behavior.  I have been in communication with their devs and created an issue on their Amplify iOS repo linked here: https://github.com/aws-amplify/amplify-ios/issues/1083.  

Once they fix the issue and we update our Amplify iOS Library dependencies, your problem will be resolved.  I don't know how long this will take on their side but can provide updates as we investigate/triage it.

**Comment by Arosf** [comment]
@fjnoyp Thank you to the corresponding!

**Comment by fjnoyp** [comment]
Hi @Arosf after looking through our code more, we have a fix in place for this issue that you reported.  

PR fix:
https://github.com/aws-amplify/amplify-flutter/pull/439

I'll reopen this issue and mark it as pending this PR.

**Comment by HuiSF** [comment]
The fix for this issue has been released with version [`0.2.3`](https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/CHANGELOG.md#023-2021-09-09). When schema changes, DataStore will automatically clear local DB and recreate tables with the latest schema and resync data from cloud.
